yt/fields/field_functions.py

```diff
@@ -7,37 +7,34 @@
 
 
 def get_radius(data, field_prefix, ftype):
-    unit_system = data.ds.unit_system
-    center = data.get_field_parameter("center").in_base(unit_system.name)
-    DW = (data.ds.domain_right_edge - data.ds.domain_left_edge).in_base(
-        unit_system.name
-    )
-    # This is in cm**2 so it can be the destination for our r later.
+    center = data.get_field_parameter("center").to("code_length")
+    DW = (data.ds.domain_right_edge - data.ds.domain_left_edge).to("code_length")
+    # This is in code_length so it can be the destination for our r later.
     radius2 = data.ds.arr(
-        np.zeros(data[ftype, field_prefix + "x"].shape, dtype="float64"), "cm**2"
+        np.zeros(data[ftype, field_prefix + "x"].shape, dtype="float64"), "code_length"
     )
-    r = radius2.copy()
+
+    r = radius2.v
```

Comment:
Ah yes. I thought that too at first but found that it changes the resulting radius. 

Both r and r2 are used in the loop below. 
r is used as intermediate storage for x^2, y^2 and z^2 and r2 to accumulate the sum of them.

Reply:
isn't the first time it's used on line 21, where it's used as a result storage array ? if it _does_ affect the result then could I see an example ? I can't quite convince myself

Reply:
Perhaps think in the old/current version, the way it's referenced with `rdw` and then how `rdw` is used as a destination in the periodicity subtraction causes it?

Reply:
but `rdw` is also a copy 🙃 
I really don't get it, but it's required that I before we can merge this :-)

Reply:
Oh I see what you are saying. 
Initially I had 
`r = radius2.d`
That gave a different result. That was because then r and radius2 pointed to the same array. 

When we do 
`r = radius2.v` in line 17
we are just allocating the extra array needed. 

If we remove the line as you suggest than r will get allocated the first time round in line 21 and we should get the same result. In fact we then did not need it on line 17. 

However, it will not change performance and define the new array implicitly. 

Perhaps we could use `r = np.zeros_like(radius.d)` in 17 to make clear that we are just allocating temporary storage here?

Reply:
So long story short your suggestion of using `np.empty` is excellent and disregard all my comments in between.

Reply:
thanks. I just pushed a similar commit, but used `np.empty_like` instead of `np.empty` as you suggested in private.
I also report that it shaves off about 1.5s more (out of 60), as compared to the previous iteration, so that'd be about an additional 2% gain.